Bug: addGCPArtifactRegistryServiceAccount TypeScript interface mismatch

## Bug Report

### Description
The `addGCPArtifactRegistryServiceAccount` helper works in JavaScript but has incorrect TypeScript interface definitions.

### Steps to Reproduce
1. Create TypeScript file with:
```typescript
rutter.addGCPArtifactRegistryServiceAccount({
  name: 'test',
  gcpServiceAccount: 'test@project.iam.gserviceaccount.com'
});
```

2. Run with ts-node

### Expected Behavior
Should compile without TypeScript errors

### Actual Behavior
```
error TS2353: Object literal may only specify known properties, and 'gcpServiceAccount' does not exist in type 'GCPArtifactRegistryServiceAccountSpec'.
```

### Environment
- Timonel version: 2.7.0
- TypeScript: Latest
- Node.js: 20+

### Workaround
Works fine in JavaScript, only TypeScript interface is incorrect.

### Fix Needed
Update TypeScript interface to match actual implementation.

--- a/src/index.ts
+++ b/src/index.ts
@@ -9,6 +9,66 @@ export * from './lib/helm.js';
 export * from './lib/umbrella.js';
 export * from './lib/security.js';
 
+// Export type definitions from resource providers
+export type {
+  DeploymentSpec,
+  DaemonSetSpec,
+  StatefulSetSpec,
+  RoleSpec,
+  ClusterRoleSpec,
+  RoleBindingSpec,
+  ClusterRoleBindingSpec,
+  JobSpec,
+  ServiceSpec,
+  ConfigMapSpec,
+  SecretSpec,
+  ServiceAccountSpec,
+} from './lib/resources/core/coreResources.js';
+
+export type {
+  PersistentVolumeSpec,
+  PersistentVolumeClaimSpec,
+  StorageClassSpec,
+} from './lib/resources/core/storageResources.js';
+
+export type {
+  AWSEBSStorageClassSpec,
+  AWSEFSStorageClassSpec,
+  AWSIRSAServiceAccountSpec,
+  AWSECRServiceAccountSpec,
+  AWSALBIngressSpec,
+} from './lib/resources/cloud/aws/awsResources.js';
+
+export type {
+  AzureDiskStorageClassSpec,
+  AzureFileStorageClassSpec,
+  AzureAGICIngressSpec,
+  AzureKeyVaultSecretProviderClassSpec,
+  AzureACRServiceAccountSpec,
+} from './lib/resources/cloud/azure/azureResources.js';
+
+export type {
+  GCPPersistentDiskStorageClassSpec,
+  GCPFilestoreStorageClassSpec,
+  GCPGCEIngressSpec,
+  GCPWorkloadIdentityServiceAccountSpec,
+  GCPArtifactRegistryServiceAccountSpec,
+} from './lib/resources/cloud/gcp/gcpResources.js';
+
+export type {
+  HorizontalPodAutoscalerSpec,
+  VerticalPodAutoscalerSpec,
+  CronJobSpec,
+} from './lib/resources/autoscaling/autoscalingResources.js';
+
+export type { NetworkPolicySpec, IngressSpec } from './lib/resources/network/networkResources.js';
+
+export type {
+  KarpenterNodePoolSpec,
+  KarpenterNodeClaimSpec,
+  KarpenterEC2NodeClassSpec,
+} from './lib/resources/cloud/aws/karpenterResources.js';
+
 // Re-export specific items from new modules to avoid conflicts
 export {
   generateHelpersTemplate,

--- a/src/lib/resources/cloud/gcp/gcpResources.ts
+++ b/src/lib/resources/cloud/gcp/gcpResources.ts
@@ -219,8 +219,15 @@ export class GCPResources extends BaseResourceProvider {
    * @since 2.7.0
    */
   addArtifactRegistryServiceAccount(spec: GCPArtifactRegistryServiceAccountSpec): ApiObject {
+    // Support both googleServiceAccount and gcpServiceAccount for backward compatibility
+    const googleServiceAccount = spec.googleServiceAccount || spec.gcpServiceAccount;
+
+    if (!googleServiceAccount) {
+      throw new Error('Either googleServiceAccount or gcpServiceAccount must be provided');
+    }
+
     const annotations = {
-      'iam.gke.io/gcp-service-account': spec.googleServiceAccount,
+      'iam.gke.io/gcp-service-account': googleServiceAccount,
       ...(spec.annotations || {}),
     };
 
@@ -295,7 +302,8 @@ export interface GCPWorkloadIdentityServiceAccountSpec {
 
 export interface GCPArtifactRegistryServiceAccountSpec {
   name: string;
-  googleServiceAccount: string;
+  googleServiceAccount?: string;
+  gcpServiceAccount?: string;
   automountServiceAccountToken?: boolean;
   imagePullSecrets?: Array<{ name: string }>;
   labels?: Record<string, string>;

--- a/src/lib/resources/cloud/gcp/gcpResources.ts
+++ b/src/lib/resources/cloud/gcp/gcpResources.ts
@@ -219,12 +219,8 @@ export class GCPResources extends BaseResourceProvider {
    * @since 2.7.0
    */
   addArtifactRegistryServiceAccount(spec: GCPArtifactRegistryServiceAccountSpec): ApiObject {
-    // Support both googleServiceAccount and gcpServiceAccount for backward compatibility
-    const googleServiceAccount = spec.googleServiceAccount || spec.gcpServiceAccount;
-
-    if (!googleServiceAccount) {
-      throw new Error('Either googleServiceAccount or gcpServiceAccount must be provided');
-    }
+    const googleServiceAccount =
+      'googleServiceAccount' in spec ? spec.googleServiceAccount : spec.gcpServiceAccount;
 
     const annotations = {
       'iam.gke.io/gcp-service-account': googleServiceAccount,
@@ -300,12 +296,13 @@ export interface GCPWorkloadIdentityServiceAccountSpec {
   annotations?: Record<string, string>;
 }
 
-export interface GCPArtifactRegistryServiceAccountSpec {
+export type GCPArtifactRegistryServiceAccountSpec = {
   name: string;
-  googleServiceAccount?: string;
-  gcpServiceAccount?: string;
   automountServiceAccountToken?: boolean;
   imagePullSecrets?: Array<{ name: string }>;
   labels?: Record<string, string>;
   annotations?: Record<string, string>;
-}
+} & (
+  | { googleServiceAccount: string; gcpServiceAccount?: never }
+  | { gcpServiceAccount: string; googleServiceAccount?: never }
+);

